### PR TITLE
Add tests and fix if conditions in TourOfferService, add new custom Exception for not found entities

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/com/epam/service/TourOfferService.java [hunk 1]
```diff
@@ -27,29 +27,29 @@ public TourOfferService(TourOfferDAO tourOfferDAO){
     }
 
     public int deleteTour(Integer tourId){
-        if(tourId!=null&&tourId!=0)
-            return tourOfferDAO.deleteTour(tourId);
-        else {
+        if (tourId == null || tourId == 0) {
```
Comment (goponenkoev) [inline src/main/java/com/epam/service/TourOfferService.java:30]: What's wrong with `0`?

File: src/test/java/integration/TourOfferServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        Assert.assertEquals(1, actualReturnCode);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addTourNull() {
+        tourOfferService.addTour(null);
+    }
+
+//    @Test(expected = IllegalArgumentException.class)
+//    public void addTourOfferWrongHotelId() {
```
Comment (goponenkoev) [inline src/test/java/integration/TourOfferServiceTest.java:79]: Oh, no. please. Let's do not commit commented code

File: src/main/java/com/epam/service/TourOfferService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/com/epam/service/TourOfferService.java:30]: I believe we dont want to save 0 id... no?

File: src/test/java/integration/TourOfferServiceTest.java (hunk 2 above)
Comment (ghost) [reply src/test/java/integration/TourOfferServiceTest.java:79]: These tests should be uncommented after fix bugs

Comment (ghost) [reply src/test/java/integration/TourOfferServiceTest.java:79]: I add tasks on the board and write tests now to test them in future

File: src/main/java/com/epam/service/TourOfferService.java (hunk 1 above)
Comment (goponenkoev) [reply src/main/java/com/epam/service/TourOfferService.java:30]: I think `id ` is auto-generated by H2

**Pull request merged**